### ISSUE TITLE
[java] A false negative about the rule ForLoopCanBeForeach

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#4094]
**Affects PMD Version:6.47.0** 

**Rule:ForLoopCanBeForeach**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#forloopcanbeforeach>

**Description:**
Hi, I found a false negative about the rule ForLoopCanBeForeach. In code example `loop1` below, PMD should have reported warnings in line 4 or line 8, but it does not. However, it can detect the bug if statements in else branch are modified, or even removedm, just like the method `loop2` below. Hence, I think this is a false negative

**Code Sample demonstrating the issue:**

#### loop1
```java
void loop1(List<String> l, boolean tag) {
    int i = 0;
    if (tag) {
        for (; i < l.size(); i++) {
            System.out.println(l.get(i));  // should report a warning, but not
        }
    } else {
        for (; i < l.size(); i++) {
            System.out.println(l.get(i));  // no warnings, too
        }
    }
}
```
#### loop2
```java
void loop2(List<String> l, boolean tag) {
    int i = 0;
    if (tag) {
        for (; i < l.size(); i++) {
            System.out.println(l.get(i));  // can report a warning
        }
    } else {
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Zustin** [comment]
I found a very neat, but maybe useful false negative case:
```java
void loop(List<String> l) {
  int i = 0;
  for (; i < l.size(); i++) {
      System.out.println(l.get(i));  // cannot detect, no warnings
  }
  i = 0;
  for (; i < l.size(); i++) {
      System.out.println(l.get(i));  // cannot detect, no warnings
  }
}
```
Two consecutive buggy loop statements cannot be detected.
